platform/pulp/platform/models/repository.py

```diff
@@ -1,10 +1,329 @@
+
+    :cvar last_content_added: When content was last added.
+    :type last_content_added: models.DateTimeField
+
+    :cvar last_content_removed: When content was last removed.
+    :type last_content_removed: models.DateTimeField
+
+    Relations:
+
+    :cvar scratchpad: Arbitrary information stashed on the repository.
+    :type scratchpad: fields.GenericRelation
+
+    :cvar notes: Arbitrary repository properties.
+    :type notes: fields.GenericRelation
+
+    :cvar content: Associated content.
+    :type content: models.ManyToManyField
+    """
+    name = models.TextField(db_index=True, unique=True)
+    description = models.TextField(blank=True)
+
+    last_content_added = models.DateTimeField(blank=True, null=True)
+    last_content_removed = models.DateTimeField(blank=True, null=True)
+
+    scratchpad = fields.GenericRelation(Scratchpad)
+    notes = fields.GenericRelation(Notes)
+
+    content = models.ManyToManyField('Content', through='RepositoryContent')
+
+    @property
+    def content_summary(self):
+        """
+        The contained content summary.
+
+        :return: A dict of {<type>: <count>}
+        :rtype:  dict
+        """
+        mapping = self.content.values('type').annotate(count=models.Count('type'))
+        return {m['type']: m['count'] for m in mapping}
+
+
+class RepositoryGroup(Model):
+    """
+    A group of repositories.
+
+    Fields:
+
+    :cvar name: The group name.
+    :type name: models.TextField
+
+    :cvar: description: An optional description.
+    :type: models.TextField
+
+    Relations:
+    :cvar notes: Arbitrary group properties.
+    :type notes: fields.GenericRelation
+
+    :cvar members: Repositories associated with the group.
+    :type members: models.ManyToManyField
+    """
+    name = models.TextField(db_index=True, unique=True)
+    description = models.TextField(blank=True)
+
+    members = models.ManyToManyField('Repository')
+    scratchpad = fields.GenericRelation(Scratchpad)
+    notes = fields.GenericRelation(Notes)
+
+
+class Plugin(MasterModel):
+    """
+    An Abstract model for plugins.
+
+    Fields:
+
+    :cvar name: The plugin name.
+    :type type: models.TextField
+
+    :cvar type: The plugin type.
+    :type type: models.TextField
+
+    :cvar last_updated: When the plugin was last updated.
+    :type last_updated: fields.DateTimeField
+
+    Relations:
+
+    """
+    name = models.TextField(db_index=True)
+    type = models.TextField(blank=False, default=None)
+    last_updated = models.DateTimeField(auto_now=True, blank=True, null=True)
+
+    class Meta:
+        abstract = True
+
+
+class Importer(Plugin):
+    """
+    A abstract content importer.
+
+    Fields:
+
+    :cvar feed_url: The URL of an external content source.
+    :type feed_url: models.TextField
+
+    :cvar validate: Validate the imported context.
+    :type validate: models.BooleanField
+
+    :cvar ssl_ca_certificate: A PEM encoded CA certificate used for authentication.
```

Comment:
This should probably read "A PEM encoded CA certificate used to verify the signature of the TLS certificate served by the ``feed_url``" or something like that or it might be confused with the CA certificate that signed the client certificates.